driver/pace/driver/registry.py

```diff
@@ -115,7 +115,8 @@ def from_dict(self, config: dict) -> T:
         It can also have a "config" key, which is a dict used to initialize the
         dataclass. By default this is an empty dict.
         """
-        config.setdefault("config", {})
+        if "config" not in config:
+            config.setdefault("config", {})
```

Comment:
`setdefault` only modifies the dictionary if the key does not exist, the if statement would be redundant.
```suggestion
        config.setdefault("config", {})
```

Reply:
Didn't realize that, thanks!